<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Start</class>
 <widget class="QMainWindow" name="Start">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>773</width>
    <height>712</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Pattern Gen</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QHBoxLayout" name="horizontalLayout_2">
    <item>
     <layout class="QVBoxLayout" name="verticalLayout_5">
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout">
        <item>
         <layout class="QVBoxLayout" name="verticalLayout">
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_7">
            <item>
             <widget class="QLabel" name="label">
              <property name="text">
               <string>Mask:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="mask"/>
            </item>
            <item>
             <widget class="QLabel" name="mask_len">
              <property name="text">
               <string>[0]</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <widget class="QLabel" name="label_2">
            <property name="text">
             <string>Custom Charsets</string>
            </property>
           </widget>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_3">
            <item>
             <widget class="QLabel" name="label_3">
              <property name="text">
               <string>1:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="custom1"/>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_4">
            <item>
             <widget class="QLabel" name="label_4">
              <property name="text">
               <string>2:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="custom2"/>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_5">
            <item>
             <widget class="QLabel" name="label_5">
              <property name="text">
               <string>3:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="custom3"/>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_6">
            <item>
             <widget class="QLabel" name="label_6">
              <property name="text">
               <string>4:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="custom4"/>
            </item>
           </layout>
          </item>
          <item>
           <widget class="Line" name="line">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_8">
            <item>
             <layout class="QVBoxLayout" name="verticalLayout_10">
              <item>
               <widget class="QCheckBox" name="use_inst">
                <property name="text">
                 <string/>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QVBoxLayout" name="verticalLayout_4">
              <item>
               <widget class="QLabel" name="label_8">
                <property name="text">
                 <string>Num of Instances</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLineEdit" name="num_inst"/>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QVBoxLayout" name="verticalLayout_3">
              <item>
               <widget class="QLabel" name="label_7">
                <property name="text">
                 <string>Num of Duplicates per Instance</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLineEdit" name="num_dupl"/>
              </item>
             </layout>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_15">
            <item>
             <layout class="QVBoxLayout" name="verticalLayout_11">
              <item>
               <widget class="QCheckBox" name="use_count">
                <property name="text">
                 <string/>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QVBoxLayout" name="verticalLayout_6">
              <item>
               <widget class="QLabel" name="label_11">
                <property name="text">
                 <string>Max lowercase</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLineEdit" name="count_l"/>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QVBoxLayout" name="verticalLayout_7">
              <item>
               <widget class="QLabel" name="label_12">
                <property name="text">
                 <string>Max uppercase</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLineEdit" name="count_u"/>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QVBoxLayout" name="verticalLayout_8">
              <item>
               <widget class="QLabel" name="label_13">
                <property name="text">
                 <string>Max digits</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLineEdit" name="count_d"/>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QVBoxLayout" name="verticalLayout_9">
              <item>
               <widget class="QLabel" name="label_14">
                <property name="text">
                 <string>Max special</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLineEdit" name="count_s"/>
              </item>
             </layout>
            </item>
           </layout>
          </item>
          <item>
           <widget class="Line" name="line_2">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_9">
            <item>
             <widget class="QLabel" name="label_9">
              <property name="text">
               <string>Save Path</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="path"/>
            </item>
            <item>
             <widget class="QPushButton" name="sel_path">
              <property name="text">
               <string>Select</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_16">
            <item>
             <widget class="QCheckBox" name="setStart">
              <property name="text">
               <string>Start from:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="startFrom"/>
            </item>
            <item>
             <widget class="QLabel" name="startFrom_len">
              <property name="text">
               <string>[0]</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_14">
            <item>
             <widget class="QCheckBox" name="threading">
              <property name="text">
               <string>Use multithreading (if save to file 2x disk space will be needed during calculation)</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_10">
            <item>
             <widget class="QCheckBox" name="multi">
              <property name="text">
               <string>Save to multiple files with size of</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="multi_size"/>
            </item>
            <item>
             <widget class="QLabel" name="label_10">
              <property name="text">
               <string>MB</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_13">
            <item>
             <widget class="QCheckBox" name="out">
              <property name="text">
               <string>Send all combinations to STDOUT</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_11">
            <item>
             <widget class="QPushButton" name="start">
              <property name="text">
               <string>Start</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="pause">
              <property name="text">
               <string>Pause</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="stop">
              <property name="text">
               <string>Stop</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_12">
            <item>
             <widget class="QPushButton" name="precalc">
              <property name="text">
               <string>Show Precalc</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="saveSession">
              <property name="text">
               <string>Save Session</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="loadSession">
              <property name="text">
               <string>Load Session</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <widget class="QLabel" name="status">
            <property name="text">
             <string/>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="verticalSpacer">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_2">
          <item>
           <widget class="QTextBrowser" name="log"/>
          </item>
          <item>
           <widget class="QProgressBar" name="progress">
            <property name="value">
             <number>24</number>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QLabel" name="statusBar">
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>773</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
